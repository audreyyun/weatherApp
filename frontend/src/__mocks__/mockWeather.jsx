const mockWeather = { 
    data: {

        "coord": {
            "lon": -118.2437,
            "lat": 34.0522
        },
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "clear sky",
                "icon": "01d"
            }
        ],
        "base": "stations",
        "main": {
            "temp": 294.01,
            "feels_like": 292.88,
            "temp_min": 290.85,
            "temp_max": 297.6,
            "pressure": 1012,
            "humidity": 28
        },
        "visibility": 10000,
        "wind": {
            "speed": 2.68,
            "deg": 215,
            "gust": 0.45
        },
        "clouds": {
            "all": 1
        },
        "dt": 1634077448,
        "sys": {
            "type": 1,
            "id": 3694,
            "country": "US",
            "sunrise": 1634046948,
            "sunset": 1634088154
        },
        "timezone": -25200,
        "id": 5368361,
        "name": "Los Angeles",
        "cod": 200
    }
}

export default mockWeather;