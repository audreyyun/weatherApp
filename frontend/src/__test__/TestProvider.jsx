import React from 'react'
import { Provider } from 'react-redux';
import configureStore from '../store/store';

const store = configureStore();

const TestProvider = ({ children }) => (
    <Provider store={store}>
        {children}
    </Provider>
);

export default TestProvider;