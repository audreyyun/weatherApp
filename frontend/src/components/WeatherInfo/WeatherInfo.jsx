import React from 'react';

const WeatherInfo = ({ weather }) => { 
    const currentWeather = weather;
    const temperatures = weather ? weather.main: '';
    const weatherData = weather.weather ? weather.weather[0] : '';

    const convertTemperature = k => {
        const fahrenheit = (1.8 * (k - 273) + 32).toFixed(2);
        const celsius = (k - 273).toFixed(2);

        return {
            fahrenheit,
            celsius
        };
    };

    const testingWeatherObject = { 
        icon: weatherData.icon,
        description: weatherData.description,
        main: weatherData.main,
        city: weather.name,
        currentTemp: convertTemperature(temperatures?.temp).fahrenheit,
        feels_like: convertTemperature(temperatures?.feels_like).fahrenheit,
        minTemp: convertTemperature(temperatures?.temp_min).fahrenheit,
        maxTemp: convertTemperature(temperatures?.temp_max).fahrenheit
    };

    const iconSrc = `http://openweathermap.org/img/wn/${testingWeatherObject.icon}@4x.png`;

    if (!weatherData) { 
        return (
            <>
                <div className="no-weather-info">Please enter a zip code and choose a country to see today's weather!</div>
            </>
        )
    } else { 
        return (
            <div className="weather-info-container">
                <div className="weather-info">
                    <h1 className="weather-info-city">{testingWeatherObject.city}</h1>
                    <h3>{testingWeatherObject.description}</h3>
                    <img src={iconSrc} alt="weather icon" />
                    <p>Current Temperature: {testingWeatherObject.currentTemp} 째F</p>
                    <br />
                    <p>Feels Like: {testingWeatherObject.feels_like} 째F</p>
                    <p>Min Temperature: {testingWeatherObject.minTemp} 째F</p>
                    <p>Max Temperature: {testingWeatherObject.maxTemp} 째F</p>
                </div>
            </div>
        )
    };
};

export default WeatherInfo;