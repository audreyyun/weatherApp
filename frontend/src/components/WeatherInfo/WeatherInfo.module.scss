@import '../../scss/style.scss';

.weatherInfoContainer { 
    display: flex;
    justify-content: center;
    padding-top: 20px;

}

.noWeatherInfo { 
    display: flex;
    justify-content: center;
    padding: 2em 10px;
    margin: 0 auto;

    font-size: 20px;
    color:rgb(255, 250, 158);
}

.weatherInfo { 
    width: 550px;
    height: 600px;
    padding: 2em 4em;
    border-radius: 1em;
    box-shadow: rgba(0, 0, 0, 0.25) 0px 5px 15px;
    line-height: 1.5em;

    .weatherInfoCity { 
        display: flex;
        justify-content: center;
        font-size: 3em;
        color:rgb(236, 239, 255);
        line-height: 1.5em;
    }
    
    h3, p, img { 
        display: flex;
        justify-content: center;
        margin: 0 auto;
        
        color:rgb(236, 239, 255);
        line-height: inherit;
    }

    h3 {
        font-size: 12px;
        line-height: 1em;
    }

}