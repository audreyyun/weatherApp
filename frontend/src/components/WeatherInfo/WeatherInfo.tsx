import React from 'react'; 
import styles from './WeatherInfo.module.scss';

type Props = { 
    weather: {
        // main: any;
        data: { 
            base: string;
            clouds: {
                all: number
            };
            cod: number;
            coord: {
                lon: number;
                lat: number;
            };
            dt: number;
            id: number;
            main: {
                temp: number;
                feels_like: number;
                temp_min: number;
                temp_max: number;
                pressure: number
            };
            name: string;
            sys: {
                type: number;
                id: number;
                country: string;
                sunrise: number;
                sunset: number;
            };
            timezone: number;
            visibility: number;
            weather: CurrentWeather[]; //notes it will be an array of type WeatherType objects
            wind: {
                deg: number;
                speed: number
            }
        }
    }
}

type CurrentWeather = { 
    icon: string | null;
    description: string | null;
    main: string | null;
}

const WeatherInfo = ({ weather } : Props) => { 
    console.log("this is weatherinfo", weather)
    const { data } = weather
    // const temperatures = weather ? weather.data?.main: '';
    // const weatherData = weather?.data ? weather.data.weather[0] : '';
    const temperatures = data?.main;
    const weatherData = data?.weather[0];
    const convertTemperature = (k: number) => {
        const fahrenheit = (1.8 * (k - 273) + 32).toFixed(2);
        const celsius = (k - 273).toFixed(2);

        return {
            fahrenheit,
            celsius
        };
    };
    const testingWeatherObject = { 
        icon: weatherData?.icon,
        description: weatherData?.description,
        main: weatherData?.main,
        city: data?.name,
        currentTemp: convertTemperature(temperatures?.temp).fahrenheit,
        feels_like: convertTemperature(temperatures?.feels_like).fahrenheit,
        minTemp: convertTemperature(temperatures?.temp_min).fahrenheit,
        maxTemp: convertTemperature(temperatures?.temp_max).fahrenheit
    };

    const iconSrc = `http://openweathermap.org/img/wn/${testingWeatherObject.icon}@4x.png`;

    if (!weatherData) { 
        return (
            <>
                <div className="no-weather-info">Please enter a city to see today's weather!</div>
            </>
        )
    } else { 
        return (
            <div className={styles.weatherInfoContainer}>
                <div className={styles.weatherInfo}>
                    <h1 className={styles.weatherInfoCity}>{testingWeatherObject.city}</h1>
                    <h3>{testingWeatherObject.description}</h3>
                    <img src={iconSrc} alt="weather icon" />
                    <p>Current Temperature: {testingWeatherObject.currentTemp} 째F</p>
                    <br />
                    <p>Feels Like: {testingWeatherObject.feels_like} 째F</p>
                    <p>Min Temperature: {testingWeatherObject.minTemp} 째F</p>
                    <p>Max Temperature: {testingWeatherObject.maxTemp} 째F</p>
                </div>
            </div>
        )
    };
};

export default WeatherInfo;