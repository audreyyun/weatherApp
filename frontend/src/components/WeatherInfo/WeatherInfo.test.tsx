import '@testing-library/jest-dom/extend-expect';

import React from 'react'
import WeatherInfo from './WeatherInfo'
import { render } from '@testing-library/react'
import TestProvider from '../../__test__/TestProvider';
// eslint-disable-next-line jest/no-mocks-import
import mockWeather from '../../__mocks__/mockWeather';


// jest.mock('./mockWeather', () => {
    
//     return {
//         mockWeather: () => { 
//             return mockWeather
//         }
//     }
// })
const WeatherInfoTest = () => (
    <TestProvider>
        <WeatherInfo weather={mockWeather} />
    </TestProvider>
);


describe('WeatherInfo', () => {
    it('should render properly', () => {
        const component = render(<WeatherInfoTest />)
        expect(component.asFragment()).toMatchSnapshot()
    })
})
