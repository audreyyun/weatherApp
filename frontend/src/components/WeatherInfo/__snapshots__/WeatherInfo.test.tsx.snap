// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo should render properly 1`] = `
<DocumentFragment>
  <div
    class="weatherInfoContainer"
  >
    <div
      class="weatherInfo"
    >
      <h1
        class="weatherInfoCity"
      >
        Los Angeles
      </h1>
      <h3>
        clear sky
      </h3>
      <img
        alt="weather icon"
        src="http://openweathermap.org/img/wn/01d@4x.png"
      />
      <p>
        Current Temperature: 69.82 째F
      </p>
      <br />
      <p>
        Feels Like: 67.78 째F
      </p>
      <p>
        Min Temperature: 64.13 째F
      </p>
      <p>
        Max Temperature: 76.28 째F
      </p>
    </div>
  </div>
</DocumentFragment>
`;
